Jenkinsfile

```diff
@@ -1,158 +1,64 @@
-    capabilities.putAll(desiredCapabilities)
-    def json = JsonOutput.toJson([capabilities: capabilities])
-    writeFile file: 'capabilities.json', text: json
-}
-
-/** Run Tox
- *
- * @param environment test environment to run
-*/
-def runTox(environment) {
-  try {
-    wrap([$class: 'AnsiColorBuildWrapper']) {
-      withCredentials([[
-        $class: 'StringBinding',
-        credentialsId: 'SAUCELABS_API_KEY',
-        variable: 'SAUCELABS_API_KEY']]) {
-        withEnv(["PYTEST_ADDOPTS=${PYTEST_ADDOPTS} " +
-          "--driver=SauceLabs " +
-          "--variables=capabilities.json " +
-          "--color=yes"]) {
-          sh "tox -e ${environment}"
-        }
-      }
-    }
-  } catch(err) {
-    currentBuild.result = 'FAILURE'
-    throw err
-  } finally {
-    dir('results') {
-      stash environment
-    }
+pipeline {
+  agent any
+  options {
+    ansiColor('xterm')
+    timestamps()
+    timeout(time: 1, unit: 'HOURS')
   }
-}
-
-/** Send a notice to #fxtest-alerts on irc.mozilla.org with the build result
- *
- * @param result outcome of build
-*/
-def ircNotification(result) {
-  def nick = "fxtest${BUILD_NUMBER}"
-  def channel = '#fx-test-alerts'
-  result = result.toUpperCase()
-  def message = "Project ${JOB_NAME} build #${BUILD_NUMBER}: ${result}: ${BUILD_URL}"
-  node {
-    sh """
-        (
-        echo NICK ${nick}
-        echo USER ${nick} 8 * : ${nick}
-        sleep 5
-        echo "JOIN ${channel}"
-        echo "NOTICE ${channel} :${message}"
-        echo QUIT
-        ) | openssl s_client -connect irc.mozilla.org:6697
-    """
+  environment {
+    VARIABLES = credentials('MOZILLIANS_VARIABLES')
+    SAUCELABS_API_KEY = credentials('SAUCELABS_API_KEY')
   }
-}
-
-stage('Checkout') {
-  node {
-    timestamps {
-      deleteDir()
-      checkout scm
-      stash 'workspace'
-    }
-  }
-}
-
-stage('Lint') {
-  node {
-    timestamps {
-      deleteDir()
-      unstash 'workspace'
-      sh 'tox -e flake8'
+  stages {
+    stage('Lint') {
+      steps {
+        sh "tox -e flake8"
+      }
     }
-  }
-}
-
-try {
-  stage('Test') {
-    node {
-      timeout(time: 1, unit: 'HOURS') {
-        timestamps {
-          deleteDir()
-          unstash 'workspace'
-          try {
-            writeCapabilities(capabilities)
-            withCredentials([[
-              $class: 'FileBinding',
-              credentialsId: 'MOZILLIANS_VARIABLES',
-              variable: 'VARIABLES']]) {
-              withEnv(["PYTEST_ADDOPTS=--variables=${VARIABLES}"]) {
-                runTox(environment)
-              }
-            }
-          } catch(err) {
-            currentBuild.result = 'FAILURE'
-            throw err
-          } finally {
-            dir('results') {
-              stash environment
-            }
-          }
+    stage('Test') {
+      environment {
+        PYTEST_ADDOPTS = "--color=yes --driver=SauceLabs --variables=capabilities.json --variables=${VARIABLES}"
```

Comment:
Once you've opened the two issues, please add comments so we can move this into the main `evnironment` section and make it more readable by splitting it over multiple lines.